cc/dual_net/factory.cc

```diff
@@ -91,8 +88,9 @@ class RemoteDualNetFactory : public DualNetFactory {
       fputc('\n', stderr);
     });
 
+    int games_per_inference = std::max(1, parallel_games / 2);
```

Comment:
I'm commenting blindly and might misunderstand the purpose of this flag.
I found that having running a couple extra parallel_games worked much better (e.g. use parallel_games / 2 -2)
Otherwise you are waiting for the 100% percentile latency over X games V virtual losses (one game on one slow thread with one tricky position keeps the entire inference batch from starting)

Reply:
This isn't a problem in our case: when a game ends, we start a new one within a couple of seconds.

Reply:
Does everything pause while we do?

if we play 32 games, a game takes 15m, this will happen ~twice a minute, so, it'll happen ~30 times in a game.

if the pause is just a few millis, no big deal though.

Reply:
Setting parallel_games > 2 * game_per_inference will also cause each game to take longer to finish though. I've not spent too much time tuning any of this, so we can most likely improve throughput with some experimentation in this area.

Ideally we'd enable these settings to be configured by the flag file and tune the numbers on GCE while running a few hundred selfplay jobs, so we can get real-world performance numbers. Sadly, that requires more work because we'd need to introduce a second mechanism to pass the config options over to the inference_worker in some coordinated fashion (so that both processes agree on the expected batch sizes). It wouldn't be too difficult, but I don't have time to work on that right now. Maybe in a couple of weeks.

Reply:
FWIW, I have no idea how long the selfplay jobs take to write the SGF and TF examples to GCS when running on GCE. On my local machine it's 1 or 2 seconds but it should be much faster for the VMs since they're colocated.